Clarifying when to use `--warn-error-options`

runleonarun opened the pull request:
closes #7156 

## What are you changing in this pull request and why?
Exceptions page:
* Reworded the [exceptions page content](https://github.com/dbt-labs/docs.getdbt.com/pull/7168/files#diff-c253306fff86f0875722c92607ac74e4209e09572d94c1c11ff78f3d3c381a64R28-R33) to suggest they use --warn-error-options if they only want to promote jinja options

Severity page:
* Reworded [sentence](https://github.com/dbt-labs/docs.getdbt.com/pull/7168/files#diff-92205076f6f62b5d3737e562ef03e8f5c14e4ac9595174f437514528d8df369dR28) to talk about config and flag. 

Warnings page:
* Reworded [paragraph and emphasized differences](https://github.com/dbt-labs/docs.getdbt.com/pull/7168/files#diff-92205076f6f62b5d3737e562ef03e8f5c14e4ac9595174f437514528d8df369dR7-R15) between `warn-error` and `warn-error-options`
* Moved the note up and put combined version sections for easier reading. [diff here](https://github.com/dbt-labs/docs.getdbt.com/pull/7168/files#diff-92205076f6f62b5d3737e562ef03e8f5c14e4ac9595174f437514528d8df369dR30-R134)
* Added new section for `-warn-error-options`

## Checklist
<!-- Ensure your PR meets the following items. Feel free to add addition checklist items to the list if additional checks need to happen before the PR is merged, such as:
- [ ] Needs technical review
- [ ] Change base branch
- [ ] Merge on xyz date
-->
- [ ] The changes in this PR meet the [docs style guide/fundamentals](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) required for all content.
- [ ] Applied the proper versioning rules if the content is for specific dbt version(s): ([version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content)
- [ ] The content in this PR requires a dbt release note, so I added one to the [release notes page](https://docs.getdbt.com/docs/dbt-versions/dbt-cloud-release-notes).).
<!--
PRE-RELEASE VERSION OF dbt (if so, uncomment):
- [ ] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/docs/dbt-versions/core-upgrade)
-->
<!-- 
ADDING OR REMOVING PAGES (if so, uncomment):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages
-->

<!-- vercel-deployment-preview -->
---
🚀 Deployment available! Here are the direct links to the updated files:


- https://docs-getdbt-com-git-fixingwarn-errors-dbt-labs.vercel.app/reference/dbt-jinja-functions/exceptions
- https://docs-getdbt-com-git-fixingwarn-errors-dbt-labs.vercel.app/reference/global-configs/warnings
- https://docs-getdbt-com-git-fixingwarn-errors-dbt-labs.vercel.app/reference/resource-configs/severity

<!-- end-vercel-deployment-preview -->

runleonarun committed:
making sure people know when to use warn error options

runleonarun commented on website/docs/reference/dbt-jinja-functions/exceptions.md:
```diff
@@ -7,7 +7,7 @@ description: "Raise warnings/errors with the `exceptions` namespace."
 The `exceptions` namespace can be used to raise warnings and errors in dbt userspace.
 
-## raise_compiler_error
+## raise_compiler_error 
```
```suggestion
## raise_compiler_error
```

runleonarun committed:
Update website/docs/reference/dbt-jinja-functions/exceptions.md

runleonarun committed:
fixing error

mirnawong1 commented on website/docs/reference/global-configs/warnings.md:
```diff
@@ -4,7 +4,15 @@ id: "warnings"
-Turning on the `WARN_ERROR` config will convert dbt warnings into errors. Any time dbt would normally warn, it will instead raise an error. Examples include `--select` criteria that selects no resources, deprecations, configurations with no associated models, invalid test configurations, or tests and freshness checks that are configured to return warnings.
+Enabling `WARN_ERROR`config or setting the `--warn-error` flag will convert dbt warnings into errors. Any time dbt would normally warn, it will instead raise an error. Examples include `--select` criteria that selects no resources, deprecations, configurations with no associated models, invalid test configurations, or tests and freshness checks that are configured to return warnings.
+
+ You would commonly use the `--warn-error` flag to promote test warnings from `severity: warn` to errors, but it actually affects all warning types, including:
```
would we normally just use this for test warnings? for some reason, i thought it could also be used in general for all warnings since its a flag you can pass as a command

mirnawong1 replied:
would we normally just use this for test warnings? for some reason, i thought it could also be used in general for all warnings since its a flag you can pass as a command. If so, i'm not sure we should say that's its used generally to promote test warnings to errors 

```suggestion
You can use the `--warn-error` flag to promote all warnings to errors, such as:
```

mirnawong1 commented on website/docs/reference/global-configs/warnings.md:
```diff
@@ -4,7 +4,15 @@ id: "warnings"
+* Selection issues (for example, `NoNodesForSelectionCriteria`)
+* Adapter deprecation warnings (for example, `AdapterDeprecationWarning`)
+
+Consider using [`--warn-error-options`](#using-warn_error_options-for-targeted-warnings) for more targeted control over which warnings are treated as errors.
```
```suggestion
Consider using [`--warn-error-options`](#using---warn-error-options-for-targeted-warnings) for more targeted control over which warnings are treated as errors. As an example, if you'd like to only promote test warnings to errors, set that flag to `--warn-error-options '{"error": ["LogTestResult"]}'`.
```

mirnawong1 commented on website/docs/reference/dbt-jinja-functions/exceptions.md:
```diff
@@ -25,7 +25,12 @@ __Example usage__:
+Use the `exceptions.warn` method to raise a compiler warning with the provided message, but any model will still be successful and be treated as a PASS. By default, warnings will not cause dbt runs to fail. However:
+
+* If you use the `--warn-error` flag, all warnings will be promoted to errors.
+* To promote only Jinja warnings to errors (and leave other warnings alone), use `--warn-error-options`. 
```
```suggestion
* To promote only Jinja warnings to errors (and leave other warnings alone), use  `--warn-error-options`. For example, `--warn-error-options '{"error": ["JinjaLogWarning"]}'`.
```

mirnawong1 commented on website/docs/reference/resource-configs/severity.md:
```diff
@@ -22,7 +22,12 @@ Here's how those play in practice:
-Note that test warn statuses will return errors instead if the [`--warn-error`](/reference/global-configs/warnings) flag is passed. Unless dbt is told to treat warnings as errors, a test with `warn` severity will never return an error.
+By default, a test with `severity: warn` will only ever return a warning, and not cause errors. However, you can promote warnings to errors using:
+
+* `--warn-error`: Promotes **all** dbt warnings (including test warnings, Jinja warnings, deprecations, etc.)
```
using italics per style guide for emphasis and completing sentence:
```suggestion
* `--warn-error`: Promotes _all_ dbt warnings (including test warnings, Jinja warnings, deprecations, and so on.) to errors.
```

mirnawong1 commented on website/docs/reference/resource-configs/severity.md:
```diff
@@ -22,7 +22,12 @@ Here's how those play in practice:
+By default, a test with `severity: warn` will only ever return a warning, and not cause errors. However, you can promote warnings to errors using:
+
+* `--warn-error`: Promotes **all** dbt warnings (including test warnings, Jinja warnings, deprecations, etc.)
+* `--warn-error-options`: Promotes **only specific types** of warnings. 
```
```suggestion
* `--warn-error-options`: Promotes _only specific types_ of warnings. 
```

runleonarun committed:
Apply suggestions from code review

Co-authored-by: Mirna Wong <89008547+mirnawong1@users.noreply.github.com>

runleonarun committed:
Merge branch 'current' into fixingwarn-errors

mirnawong1 committed:
clarify warn-error in prod (#7169)

this pr adds a callout to inform users that they should be cautious
using warn_error/warn_error_options in prod since it can promote all
warnings into errors, including newly introduced ones.

also added some headers to distinguish content or try to provide
structure

Resolves #7130

<!-- vercel-deployment-preview -->
---
🚀 Deployment available! Here are the direct links to the updated files:


-
https://docs-getdbt-com-git-update-warn-error-dbt-labs.vercel.app/reference/global-configs/warnings

<!-- end-vercel-deployment-preview -->

---------

Co-authored-by: Grace Goheen <53586774+graciegoheen@users.noreply.github.com>
Co-authored-by: Leona B. Campbell <3880403+runleonarun@users.noreply.github.com>
Co-authored-by: runleonarun <leona.campbell@dbtlabs.com>

runleonarun committed:
Merge branch 'current' into fixingwarn-errors
